internal/controller/controllers/crd_utils.go

```diff
@@ -164,6 +167,18 @@ func (u *CRDUtils) CreateAgentCR(ctx context.Context, log logrus.FieldLogger, ho
 	return nil
 }
 
+// DoesBMHCRDExist returns true if the BareMetalHost CRD is defined in the cluster
+func (u *CRDUtils) DoesBMHCRDExist(mgr manager.Manager) bool {
```

Comment:
Thinking this function doesn't belong here... perhaps just in main since it's called once at the beginning of the program?

Reply:
+1